config/config_test.go

```diff
@@ -0,0 +1,333 @@
+				}
+
+				// Enable ACLs to test with the auth method
+				srvConfig = testutil.ConsulACLConfigFn
+				_, apiCfg := testutil.ConsulServer(t, srvConfig)
+				consulClient, err := api.NewClient(apiCfg)
+				require.NoError(t, err)
+				fakeAws := testutil.AuthMethodInit(t, consulClient, c.taskMeta.Family, c.cfg.ConsulLogin.Method)
+
+				// Use the fake local AWS server.
+				c.cfg.ConsulLogin.STSEndpoint = fakeAws.URL + "/sts"
+			}
+
+			cfg, err := c.cfg.ConsulServerConnMgrConfig(c.taskMeta)
+			require.NoError(t, err)
+
+			expectedCfg := c.expConfig(c.taskMeta)
+			if testutil.EnterpriseFlag() {
+				expectedCfg.Credentials.Login.Namespace = "test-namespace"
+				expectedCfg.Credentials.Login.Partition = "test-partition"
+			}
+
+			require.Equal(t, expectedCfg.Addresses, cfg.Addresses)
+
+			if c.cfg.ConsulLogin.Enabled {
+				require.Equal(t, expectedCfg.Credentials.Type, cfg.Credentials.Type)
+				require.Equal(t, expectedCfg.Credentials.Login.AuthMethod, cfg.Credentials.Login.AuthMethod)
+				require.Equal(t, expectedCfg.Credentials.Login.Namespace, cfg.Credentials.Login.Namespace)
+				require.Equal(t, expectedCfg.Credentials.Login.Partition, cfg.Credentials.Login.Partition)
+				require.Equal(t, expectedCfg.Credentials.Login.Meta, cfg.Credentials.Login.Meta)
+				require.Equal(t, expectedCfg.Credentials.Login.Datacenter, cfg.Credentials.Login.Datacenter)
+				require.NotEmpty(t, cfg.Credentials.Login.BearerToken)
+			}
+		})
+	}
+}
+
+func TestConsulServerConnManagerConfig_TLS(t *testing.T) {
+	caFile, err := os.CreateTemp("", "")
+	t.Cleanup(func() {
+		_ = os.RemoveAll(caFile.Name())
+	})
+	require.NoError(t, err)
+	_, err = caFile.WriteString(testCA)
+	require.NoError(t, err)
+
+	cases := map[string]struct {
+		cfg        *Config
+		setupEnv   func() error
+		cleanupEnv func() error
+	}{
+		"TLS with CACertFile": {
+			cfg: &Config{
+				ConsulServers: ConsulServers{
+					Hosts:      "consul.dc1.address",
+					EnableTLS:  true,
+					CACertFile: caFile.Name(),
+				},
+			},
+		},
+		"TLS with CACertPEM": {
+			setupEnv: func() error {
+				return os.Setenv(consulCACertPemEnvVar, testCA)
+			},
+			cfg: &Config{
+				ConsulServers: ConsulServers{
+					Hosts:     "consul.dc1.address",
+					EnableTLS: true,
+				},
+			},
+			cleanupEnv: func() error {
+				return os.Unsetenv(consulCACertPemEnvVar)
+			},
+		},
+	}
+
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			if c.setupEnv != nil {
+				require.NoError(t, c.setupEnv())
+				t.Cleanup(func() {
+					require.NoError(t, c.cleanupEnv())
+				})
+			}
+			cfg, err := c.cfg.ConsulServerConnMgrConfig(awsutil.ECSTaskMeta{})
+			require.NoError(t, err)
+
+			require.NotNil(t, cfg.TLS.RootCAs)
+		})
+	}
+}
+
+func TestClientConfig(t *testing.T) {
+	caFile, err := os.CreateTemp("", "")
+	t.Cleanup(func() {
+		_ = os.RemoveAll(caFile.Name())
+	})
+	require.NoError(t, err)
+	_, err = caFile.WriteString(testCA)
+	require.NoError(t, err)
```

Comment:
minor: we write the testCA file in a couple of places, so we could create a helper function for it.